### PR TITLE
Clarify that the CREATE MV WITH example is for Iceberg

### PR DESCRIPTION
**Pull request by jhlodin**
## Description

It's not clear from the examples given for CREATE MATERIALIZED VIEW ... WITH that their syntax is for an Iceberg catalog. Added a quick note to that effect.

> Is this change a fix, improvement, new feature, refactoring, or other?

Improvement

> Is this a change to the core query engine, a connector, client library, or the SPI interfaces? (be specific)

Documentation

> How would you describe this change to a non-technical end user or system administrator?

Clarify what catalogs you can use the examples for.

## Documentation

( ) No documentation is needed.
(x) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

(x) No release notes entries required.
( ) Release notes entries required with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: docs/src/main/sphinx/sql/create-materialized-view.rst [hunk 1]
```diff
@@ -76,8 +76,8 @@ Create or replace a materialized view ``order_totals_by_date`` that summarizes
         FROM orders
         GROUP BY orderdate;
 
-Create a materialized view with a comment and partitioning on two fields in the
-storage::
+Create a materialized view for an Iceberg catalog with a comment and
```
Comment (mosabua) [inline docs/src/main/sphinx/sql/create-materialized-view.rst:79]: lets not use Iceberg catalog ..

its a catalog using the Iceberg connector..

File: docs/src/main/sphinx/sql/create-materialized-view.rst [hunk 2]
```diff
@@ -88,7 +88,7 @@ storage::
         JOIN customer AS c
         ON o.custkey = c.custkey;
 
-Set multiple properties::
+Set multiple properties, for an Iceberg catalog::
```
Comment (mosabua) [inline docs/src/main/sphinx/sql/create-materialized-view.rst:91]: Drop this since it just repeats what was said already

**Comment by m57lyra** [comment]
@jhlodin - are all the other examples valid for all other supported sources? Do we need a generalized note saying that the syntax can differ between sources?

**Comment by jhlodin** [comment]
> @jhlodin - are all the other examples valid for all other supported sources? Do we need a generalized note saying that the syntax can differ between sources?

Technically we already have one, the last paragraph of the description says that "The supported properties are different for each connector and detailed in the SQL support section of the specific connector's documentation" which is consistent with other similar SQL statement descriptions.

**Comment by hashhar** [comment]
This doesn't scale - we'll end up adding something to the effect of the SQL Support content from each connector into the respective syntax pages over time.

It's easier for things which connect to a single system as in SQL Server docs (https://docs.microsoft.com/en-us/sql/t-sql/statements/create-table-sql-graph?view=sql-server-ver15#remarks) but that's not the case for us.

Would it be better to add a more prominent general note to all pages under `sql/` (I would argue we already mention this but people copy examples directly)?

**Comment by hashhar** [comment]
Discussed offline with @jhlodin. The problem is more about the fact that different connectors may have different table or column properties even though syntax is same. So it's important to qualify an example with what connector the example is for.

**Review by hashhar — APPROVED** [review]

**Pull request merged**